aptos-move/block-executor/src/executor.rs

```diff
@@ -861,11 +869,15 @@ where
         if let Some(reads_needing_delayed_field_exchange) =
             last_input_output.reads_needing_delayed_field_exchange(txn_idx)
         {
-            for (key, (value, layout)) in reads_needing_delayed_field_exchange.into_iter() {
-                patched_resource_write_set.insert(
-                    key,
-                    Self::replace_ids_with_values(&value, layout.as_ref(), &latest_view),
-                );
+            for (key, layout) in reads_needing_delayed_field_exchange.into_iter() {
+                if let Ok(MVDataOutput::Versioned(_, ValueWithLayout::Exchanged(value, _))) =
```

Comment:
could/should we (debug_)assert that layouts are equal?

Reply:
added TODO for @vusirikala

Reply:
yeah, they are in the wrong place, and @vusirikala  is moving the function, so leaving it for that PR